src/legacy/core_plugins/metrics/server/np/plugin.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { InternalCoreSetup } from 'kibana/server';
+// @ts-ignore
+import { fieldsRoutes } from '../routes/fields';
+// @ts-ignore
+import { visDataRoutes } from '../routes/vis';
+// @ts-ignore
+import { SearchStrategiesRegister } from '../lib/search_strategies/search_strategies_register';
+
+export class MetricsPlugin {
+  public setup(core: InternalCoreSetup) {
+    fieldsRoutes(core);
+    visDataRoutes(core);
```

Comment:
If you know you will only be needing `core.http.server`, I'd recommend only passing that down as an argument to these functions, rather than passing the entirety of `core`.

For items like these that don't rely on much of core, this will make maintenance & upgrades easier down the road -- Because I can immediately tell that `fieldsRoutes` and `visDataRoutes` only rely on `core.http.server`, without needing to look at those functions individually.

Reply:
Reverted.